server/src/internalClusterTest/java/org/elasticsearch/search/fieldcaps/FieldCapabilitiesIT.java

```diff
@@ -670,49 +675,68 @@ public void testManyIndicesWithSameMapping() {
     }
 
     public void testCancel() throws Exception {
-        BlockingOnRewriteQueryBuilder.blockOnRewrite();
-        PlainActionFuture<Response> future = PlainActionFuture.newFuture();
-        Request restRequest = new Request("POST", "/_field_caps?fields=*");
-        restRequest.setEntity(new StringEntity("""
-                  {
-                    "index_filter": {
-                        "blocking_query": {}
-                     }
-                  }
-            """, ContentType.APPLICATION_JSON.withCharset(StandardCharsets.UTF_8)));
-        Cancellable cancellable = getRestClient().performRequestAsync(restRequest, wrapAsRestResponseListener(future));
-        logger.info("--> waiting for field-caps tasks to be started");
-        assertBusy(() -> {
-            List<TaskInfo> tasks = client().admin()
-                .cluster()
-                .prepareListTasks()
-                .setActions("indices:data/read/field_caps", "indices:data/read/field_caps[n]")
-                .get()
-                .getTasks();
-            assertThat(tasks.size(), greaterThanOrEqualTo(2));
-            for (TaskInfo task : tasks) {
-                assertTrue(task.cancellable());
-                assertFalse(task.cancelled());
-            }
-        }, 30, TimeUnit.SECONDS);
-
-        cancellable.cancel();
-        logger.info("--> waiting for field-caps tasks to be cancelled");
-        assertBusy(() -> {
-            List<TaskInfo> tasks = client().admin()
-                .cluster()
-                .prepareListTasks()
-                .setActions("indices:data/read/field_caps", "indices:data/read/field_caps[n]")
-                .get()
-                .getTasks();
-            for (TaskInfo task : tasks) {
-                assertTrue(task.cancellable());
-                assertTrue(task.cancelled());
-            }
-        }, 30, TimeUnit.SECONDS);
-
-        BlockingOnRewriteQueryBuilder.unblockOnRewrite();
-        expectThrows(CancellationException.class, future::actionGet);
+        MockLogAppender logAppender = new MockLogAppender();
+        logAppender.start();
+        Logger fieldCapsLogger = LogManager.getLogger(TransportFieldCapabilitiesAction.class);
+        Loggers.addAppender(fieldCapsLogger, logAppender);
+        Level previousLogLevel = fieldCapsLogger.getLevel();
+        Loggers.setLevel(fieldCapsLogger, Level.TRACE);
```

Comment:
I think this would be better expressed as a `@TestLogging` annotation on the test.

Reply:
++, adjusted in https://github.com/elastic/elasticsearch/pull/93716/commits/d9efc1ed890207d838501981cf093a8fa4ad993d